simpleflow/task.py

```diff
@@ -49,6 +49,12 @@ class ActivityTask(Task):
     def __init__(self, activity, *args, **kwargs):
         if not isinstance(activity, Activity):
             raise TypeError('Wrong value for `activity`, got {} instead'.format(type(activity)))
+        # Keep original arguments for use in subclasses
+        # For instance this helps casting a generic class to a simpleflow.swf.task,
+        # see simpleflow.swf.task.ActivityTask.from_generic_task() factory
+        self._args = args
+        self._kwargs = kwargs
```

Comment:
Ah, yes! I moved it to the top for that reason and forgot to handle this case. But `copy()` doesn't do recursive copying on nested objects, I think? Maybe it's better to handle the `pop`'ed arguments on the specialized task in the factory? What would you prefer?